### ISSUE TITLE
Consider supporting `nogloss` environments

### ISSUE THREAD

**Issue by omentic** [retroflexivity/typst-eggs#16]
@colinbr mentioned that it'd be handy if eggs could support `nogloss` environments. 

I hadn't heard of this before, but it appears to be a thing in LaTeX's `expex`: 

<img width="666" height="550" alt="Image" src="https://github.com/user-attachments/assets/d880b4e0-194d-4f51-b1fe-26be2d25b919" />

There, the brackets overhang the sides of the gloss, without counting as the start of a word alignment. It looks like the alignment environment makes _space_ for them -- but the alignment for words of future lines starts after the brackets on the first line.

Right now, we can fake it pretty well with ``` ` ` ``` whitespace (whether used as a "dummy" element, or as padding, or to conjoin two elements:

```typst
#example[
  - Fa'nu'i  yu'  ni  [[_O_  t#text(style: "italic")[in]aitai-mu`      `_t_] na  lepblu].
  - show     me   Obl    `  `Op  _WH_[obj].read-agr  L   book
]
```

<img width="432" height="44" alt="Image" src="https://github.com/user-attachments/assets/318228f2-0ec3-4db8-b26c-dc9f91823fe3" />

Though of course this is still faking it. You've got to add ``` ` ` ``` on all the relevant lines, which isn't amazing. 

I'm not sure if there's anything better we could do with first-class support, though... this might be more of a #15 problem.

**Comment by retroflexivity** [comment]
actually, there's [hide](https://typst.app/docs/reference/layout/hide/) in typst, which allows for precise padding

```typst
#example[
  - Fa'nu'i  yu'  ni   [[_O_  t#text(style: "italic")[in]aitai-mu  _t_] na  lepblu].
  - show     me   Obl  #hide[\[\[]Op  _WH_[obj].read-agr  #hide[_t_]  L   book
]
```

<img width="627" height="84" alt="Image" src="https://github.com/user-attachments/assets/e08c3ca0-4fee-4f32-ae70-7ac57ea7ed37" />

not as cool as `\nogloss` but i'd say good enough (then it's really a #15 problem). a proper implementation would maybe look like

```typst
#let nogloss = it => {metadata((pad: [it])); it}
```

then for every gloss word, look for metadata at the start of the first / any gloss line, insert the content inside `hide` at the beginning of other lines. now, the problem is completely unglossed words like the trace above, which would require inserting a null element below it, otherwise the words would shift.

expex solves this by treating `\nogloss` as a complete word and joining it with the next via `@`, but there's no such thing as `@` in eggs. what are other use cases for it anyway? i don't remember ever using it

**Comment by omentic** [comment]
I think `hide` is capable enough! Good to rely on core Typst functionality when we can. 

Closing as this is now documented.
